src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -157,10 +157,19 @@ var ReactDOM = {
     }
     // Unsound duck typing.
     const component = (componentOrElement : any);
-    if (component.nodeType === 1) {
+    if (component.nodeType === 1) { // Element
       return component;
     }
-    return DOMRenderer.findHostInstance(component);
+    if (component.updater) { // Component
+      const whileRendering = component._reactInternalInstance.progressedChild;
+      if (!component.updater.isMounted(component) && whileRendering) {
```

Comment:
We don't need to look up the updater dynamically. We know which updater it is here. Because there is exactly one single associated with `ReactDOMFiber`. However, we shouldn't drill through the abstraction layers here. We should expose something more specific if we need to.

However, there is already an `isMounted` check (which is not free), happening inside `findHostInstance` so we shouldn't need to redo it.